main.go

```diff
@@ -94,6 +104,108 @@ func showErrorMessages(action string, errorMessages map[string]error) error {
 	return nil
 }
 
+type ConfProfile struct {
+	RemoteConfiguration bool `json:"active"`
+	Compliance          bool `json:"compliance"`
+	Insights            bool `json:"insights"`
+	Remediation         bool `json:"remediations"`
+}
+
+// MarshalJSON will convert the ConfProfile to a more meaningful structure
+func (p ConfProfile) MarshalJSON() ([]byte, error) {
+	type P struct {
+		RemoteConfiguration bool `json:"Remote Configuration"`
+		Compliance          bool `json:"Compliance"`
+		Insights            bool `json:"Insights"`
+		Remediation         bool `json:"Remediations"`
+	}
+	return json.Marshal(P(p))
+}
+
+// getConfProfile will retrieve the profile the system is configured
+func getConfProfile(client *http.Client) (*ConfProfile, error) {
+	var profile ConfProfile
+
+	// Get the server hostname where this host is connected
+	var serverHost, err = getBaseURL()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get server hostname: %w", err)
+	}
+
+	var uString string
+	// Check connection for satellite to create the final URL
```

Comment:
This checking connection to satellite server is wild. I'm sorry I don't like it.

I would just try to connect to server providing the profile. If there are more possible instances of the server providing profile, then there has to be some configuration file, where we should be able to read URL of the server.